### ISSUE TITLE
ESS_GUI: Issues with the SLD plot in the Onion Model in 5.x

### ISSUE THREAD

**Issue by smk78** [SasView/sasview#1669]
If you display the SLD plot in the Onion model, it displays, but if you click anywhere outside the SLD plot it vanishes!

Also, there is no way to access the plot control toolbar, or to save the SLD plot as data values (something requested by User MarcoH).

Other multiplicity models may be similarly affected.

**Comment by rozyczko** [comment]
The SLD plot doesn't really vanish. It gets behind the main canvas. You can bring it back by either clicking "Show SLD Profile" button again or Alt-Tabbing or clicking on the SasView icon in the taskbar and selecting the SLD profile widget.
This behaviour is similar to other tools, like most calculators and utilities, which have their windows separate from the main window.
Maybe we can make these windows always stay on top of the main app window? But then this will clutter the space even more, I'm afraid.

This isn't a bug really but more a design issue/flaw/unintended behaviour?

**Comment by rozyczko** [comment]
Lack of toggle for viewing the matplotlib chart control is an issue, though.

**Comment by smk78** [comment]
> The SLD plot doesn't really vanish. It gets behind the main canvas. You can bring it back by either clicking "Show SLD Profile" button again or Alt-Tabbing or clicking on the SasView icon in the taskbar and selecting the SLD profile widget.
> This behaviour is similar to other tools, like most calculators and utilities, which have their windows separate from the main window.
> Maybe we can make these windows always stay on top of the main app window? But then this will clutter the space even more, I'm afraid.
> 
> This isn't a bug really but more a design issue/flaw/unintended behaviour?

Oh yes, so it does. Sorry, I hadn't spotted that!

**Comment by smk78** [comment]
> Lack of toggle for viewing the matplotlib chart control is an issue, though.

And inability to access the numbers in the profile.

**Comment by butlerpd** [comment]
I wonder if the ticket to generate here is that all plots should be able to output the data in the plot as an ascii file of x,y and/or x,y,z (for 2D plots)?  Is this something available in matplotlib maybe that we can leverage?  Just a thought.

**Comment by rozyczko** [comment]
There are two types of 1D plots - standard and "quick" (in 4.x originally - I copied the behaviour to 5).
The quick plots (SLD and some other) have very limited interaction level and functionality, presumably because the additional features were not required.
I can convert the SLD plot to standard plot but this should probably be a more general issue - do we want to keep the distinction between "quick" and "standard" plots or should we only have one type?

**Comment by smk78** [comment]
Doing something, as a User was indeed trying to do, like getting hold of the values for the SLD plot so they could replot them seems to me to be a perfectly reasonable thing to do. So I would say, make all plots "standard".

**Comment by butlerpd** [comment]
I have to say that I cannot remember when or why the concept of adding a "quickplot" came from. But I would agree that it seems like general good practice that one not have too many variations on a common widget because eventually one will expect that everything I can do with say plot A I should be able to do with plot B .. it is after all just a plot (so get points, print, save to file, zoom in on etc etc)

So I would second the motion above that we should move away from the concept of a quick plot unless there is a specific and clear need to have a plot which CANNOT do the things that all plots do?

**Comment by RichardHeenan** [comment]
I agree too, but will need the sld profile to appear as yet another workspace in the data explorer panel, so maybe a larger job to do this than simply adding a plot.

**Comment by rozyczko** [comment]
> I agree too, but will need the sld profile to appear as yet another workspace in the data explorer panel, so maybe a larger job to do this than simply adding a plot.

Yes, converting quick plots to standard plots will mean they will become "proper members" of the dataset, with their own entry in the Data Explorer

**Comment by butlerpd** [comment]
I'm starting to think we may need a real "codecamp" with a breakout discussion/planning around plotting and maybe the data explorer workspace as well both of which are getting quite complicated and cluttered having grown a bit organically.  Maybe it is time to look at all the places these get used and might be used in work on a more integrated/coherent approach?  For example besides quick and standard plots we have 2D and 1d and 3D (oomf) and subplots (or should perhaps be subplots) with results from dream for example or residuals which could arguably make more sense as part of the fitting plot since they make most sense if the Q scale is exactly the same. or the thumbnail plots used in doing math on data sets. Then there is the constant complaint of being quickly drowned in plots when we are working on a SAS project (i.e. a bunch of different types of data sets at once) and the converse of not seeing the data when sending data to fitting (you have to hit show plot after having hit send to fitting) which is very disconcerting to a new person (and still is somewhat to those of us who know:-) as to the user scientist pretty much always wants to the see "the data" which is king. All other information you want when you want it but the data is where it all begins and ends :-) 

Perhaps figuring out how best to address these competing and sometimes contradictory "demands" may require more than a reactionary approach (i.e. make changes to answer a specific complaint)?  I really don't have many answers but seems to me this might be becoming ... "complicated" ?

**Comment by butlerpd** [comment]
Thinking about this some more, maybe the concept of a "quick plot" makes sense but differently than before.  I am thinking that what would make sense is that quick plots are, as we already mentioned, just plots with all the same widget tools etc BUT do not link the plotted data to the data manager and are thus ephemeral: once the plot is deleted all its content is gone (not saved as a an element in the explorer?) -- we could very quickly clutter up the data manager otherwise I fear.

**Comment by smk78** [comment]
Is there a way to retain quickplots but make the data in them available?

**Comment by rozyczko** [comment]
> Is there a way to retain quickplots but make the data in them available?

Yes! We can add a "Freeze"-like functionality to the context menu in a quick plot. This would make the plot a permanent top-level member of the Data Explorer.

**Comment by smk78** [comment]
If that is 'easy', then I'm just wondering whether that would be a good workaround, to give Users what they want, until the in-depth review @butlerpd suggests can take place?

**Comment by smk78** [comment]
User ChrisG encourages a resolution to the issue of allowing the data behind quick plots to be accessed.
